lib/client/src/account_fetcher.rs

```diff
@@ -268,7 +282,14 @@ impl<T: AccountFetcher + 'static> AccountFetcher for CachedAccountFetcher<T> {
         }
     }
 
-    fn rpc(&self) -> &RpcClientAsync {
-        self.fetcher.rpc()
+    async fn fetch_multiple_accounts(
+        &self,
+        keys: &[Pubkey],
+    ) -> anyhow::Result<Vec<(Pubkey, AccountSharedData)>> {
+        self.fetcher.fetch_multiple_accounts(keys).await
```

Comment:
I implemented `fetch_multiple_accounts` naively, so it will not fetch accounts it already has. Instead, the keeper has been configured to never use fallback oracles (otherwise there would be much trouble with the staleness checks)

Reply:
I don't think it's good to use the first requested pubkey as a key for the job. There could be difficult to find bugs when there are multiple fetch_multiple_accounts running with different lists but the same first key.

I think we either just race and allow multiple fetch jobs to run at the same time or we do something like sort+hash the account keys that are being fetched.

Since multiple fetch_multiple queries with slightly different lists would run in parallel anyway, I think it's ok to give up and just
- check cache, return if all there
- start the job for the missing keys
- put them in the cache on success
- return